### PR TITLE
feat: simplify PouchError instantiation

### PR DESCRIPTION
**Pull request by alxndrsn**

### CODE REVIEW

File: packages/node_modules/pouchdb-errors/src/index.js [hunk 1]
```diff
@@ -17,52 +17,47 @@ class PouchError extends Error {
...
-      }
-    }
+  const pouchError = new PouchError(error.status, error.name, error.message);
 
-    if (this.stack === undefined) {
-      this.stack = (new Error()).stack;
+  // inherit error properties from our parent error manually
+  // so as to allow proper JSON parsing.
+  for (const name of Object.getOwnPropertyNames(error)) {
+    if (typeof error[name] !== 'function' && name !== 'message' && name !== 'status' && name !== 'error' && name !== 'name' && name !== 'stack') {
```
Comment (alxndrsn) [inline packages/node_modules/pouchdb-errors/src/index.js:51]: simplify this

File: packages/node_modules/pouchdb-errors/src/index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class PouchError extends Error {
-  constructor(status, error, reason) {
-    super();
+  constructor(status, name, reason) {
+    super(reason);
     this.status = status;
-    this.name = error;
-    this.message = reason;
+    this.name = name;
+    // TODO this looks pointless:
```
Comment (alxndrsn) [inline packages/node_modules/pouchdb-errors/src/index.js:6]: remove this if possible

**Review by SourceR85 — APPROVED** [review]
LGTM

**Comment by craftzdog** [comment]
when will it be released? @alxndrsn

**Comment by Pierre-AlexandreTavel** [comment]
Hello,

I am in the process of updating pouchDB 8.0.1 -> 9.0.0 in an expo 52.0.37 project (React native 0.76.7), also using op-sqlite.

I am currently experiencing the following error:  (NOBRIDGE) ERROR  SQLite threw an error [TypeError: Error.stack getter called with an invalid receiver]

This error occurs when I try to replicate a database:

import HttpPouch from 'pouchdb-adapter-http';
import sqliteAdapter from 'pouchdb-adapter-react-native-sqlite';
import PouchDB from 'pouchdb-core';
import find from 'pouchdb-find';
import mapreduce from 'pouchdb-mapreduce';
import replication from 'pouchdb-replication';

const PouchDBCore = PouchDB
.plugin(HttpPouch)
.plugin(replication)
.plugin(mapreduce)
.plugin(find)
.plugin(sqliteAdapter);

this.remoteDb = new PouchDBCore(urlPlayDB + '/' + remoteDbName, {
  auth: {
    username: dbUsername,
    password: dbPassword,
  },
});
this.localDB = new PouchDBCore(localDbName + '.db', { adapter: 'react-native-sqlite' });
this.replication = this.remoteDb.replicate.to(this.localDB, finalOptions);
This error seems to be linked to this problem: https://github.com/pouchdb/pouchdb/pull/9028, so I've tried different solutions, such as using PouchDB's prerealase, applying patches from PR https://github.com/pouchdb/pouchdb/pull/9028, but the error remains.

Do you have any idea what could be causing the problem?
Thanks for your help

**Pull request merged**